src/languages/types.ts

```diff
@@ -190,7 +192,41 @@ type RemovedTheRequestParams = {valueName: string; oldValueToDisplay: string};
 
 type UpdatedTheRequestParams = {valueName: string; newValueToDisplay: string; oldValueToDisplay: string};
 
+/* Translation Object types */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type TranslationBaseValue = string | string[] | ((...args: any[]) => string);
+
+type Translation = {[key: string]: TranslationBaseValue | Translation};
+
+/* Flat Translation Object types */
+// Flattens an object and returns concatenations of all the keys of nested objects
+type FlattenObject<T, Prefix extends string = ''> = {
```

Comment:
Could you rename these generics?

- `T` to `TObject`
- `Prefix` to `TPrefix`
- `K` to `TKey`